### PR TITLE
MGMT-8067: Adds disk encryption feature flag

### PR DESCRIPTION
**Pull request by nmagnezi**
# Assisted Pull Request

## Description
Adds the Env variable DISK_ENCRYPTION_SUPPORT.
This will affect both REST-API and Kube-API.

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/cluster/validations/validations.go [hunk 1]
```diff
@@ -501,11 +501,14 @@ func ValidateIPAddressFamily(ipV6Supported bool, elements ...*string) error {
...
 }
 
-func ValidateDiskEncryptionParams(diskEncryptionParams *models.DiskEncryption) error {
+func ValidateDiskEncryptionParams(diskEncryptionParams *models.DiskEncryption, DiskEncryptionSupport bool) error {
 	if diskEncryptionParams == nil {
 		return nil
 	}
-	if diskEncryptionParams.Mode != nil && *diskEncryptionParams.Mode == models.DiskEncryptionModeTang {
+	if !DiskEncryptionSupport && swag.StringValue(diskEncryptionParams.EnableOn) != models.DiskEncryptionEnableOnNone {
+		return errors.New("disk encryption support is not enabled. Cannot apply configurations to the cluster")
```
Comment (danielerez) [inline internal/cluster/validations/validations.go:509]: Disk

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -2802,6 +2803,7 @@ func (b *bareMetalInventory) updateClusterData(_ context.Context, cluster *commo
 		b.setUsage(value, usage.SchedulableMasters, nil, usages)
 	}
 
+	logrus.Info("before params.ClusterUpdateParams.DiskEncryption != nil ")
```
Comment (danielerez) [inline internal/bminventory/inventory.go:2806]: shouldn't it be removed? or do we want to keep these for debugging?

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -2814,6 +2816,7 @@ func (b *bareMetalInventory) updateClusterData(_ context.Context, cluster *commo
 		}
 		b.setDiskEncryptionUsage(&cluster.Cluster, params.ClusterUpdateParams.DiskEncryption, usages)
 	}
+	logrus.Info("after params.ClusterUpdateParams.DiskEncryption != nil ")
```
Comment (danielerez) [inline internal/bminventory/inventory.go:2819]: same

File: cmd/main.go [hunk 4]
```diff
@@ -141,6 +141,7 @@ var Options struct {
 	EnableElasticAPM            bool          `envconfig:"ENABLE_ELASTIC_APM" default:"false"`
 	WorkDir                     string        `envconfig:"WORK_DIR" default:"/data/"`
 	LivenessValidationTimeout   time.Duration `envconfig:"LIVENESS_VALIDATION_TIMEOUT" default:"5m"`
+	EnableDiskEncryption        bool          `envconfig:"ENABLE_DISK_ENCRYPTION" default:"true"`
```
Comment (danielerez) [inline cmd/main.go:144]: why do we need it both in main and inventory? seems it's used only in the inventory Config

**Comment by nmagnezi** [comment]
@danielerez Thanks for the review. Github does not allow me to respond inline, for some reason - but I accepted and fixed all of your comments.

**Pull request merged**